### PR TITLE
[Refactor] Refactor orc create fill function logical and fix orc timestamp/timestamp instant problem

### PR DESCRIPTION
**Pull request by Smith-Cruise**
Now we create fill functions by orc::Types and TypeDescriptor.
So that we can handle orc timestamp and timestamp instant properly.

## What type of PR is this:
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [x] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

File: be/src/formats/utils.h [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <boost/algorithm/string.hpp>
+#include <string>
+
+namespace starrocks {
+class Utils {
```
Comment (dirtysalt) [inline be/src/formats/utils.h:21]: is there similar header file in starrocks repo? If there is, you don't need to create another file.

**Review by dirtysalt — APPROVED** [review]

File: be/src/formats/orc/orc_chunk_reader.cpp [hunk 2]
```diff
@@ -495,7 +478,8 @@ OrcChunkReader::~OrcChunkReader() {
     _slot_id_to_desc.clear();
     _slot_id_to_position.clear();
     _cast_exprs.clear();
-    _fill_functions.clear();
+    _column_readers.clear();
+    //    _fill_functions.clear();
```
Comment (stephen-shelby) [inline be/src/formats/orc/orc_chunk_reader.cpp:482]: delete it

**Review by stephen-shelby — APPROVED** [review]

File: be/src/formats/orc/column_reader.cpp [hunk 3]
```diff
@@ -0,0 +1,1060 @@
...
+        for (int i = col_start; i < col_start + size; ++i, ++from) {
+            values[i] = (cvbd[from] != 0);
+        }
+    }
+    return Status::OK();
+}
+
+template <LogicalType Type>
+Status IntColumnReader<Type>::get_next(orc::ColumnVectorBatch* cvb, ColumnPtr& col, size_t from, size_t size) {
+    if (_nullable) {
```
Comment (stephen-shelby) [inline be/src/formats/orc/column_reader.cpp:170]: abstract duplicated block of  if.. else..

Comment (Smith-Cruise) [reply be/src/formats/orc/column_reader.cpp:170]: refactor it step by step

**Comment by leoluan2009** [comment]
Hi @Smith-Cruise, will this pr back port to branch-2.5?

**Comment by Smith-Cruise** [comment]
> Hi @Smith-Cruise, will this pr backport to branch-2.5?

Not in my plan now, I think you can cp to 2.5 and make a pr request, I can help you to review it.

I'm afraid will introduce lots of code conflicts.

**Pull request merged**